dot/rpc/subscription/listeners.go

```diff
@@ -88,26 +97,28 @@ func (s *StorageObserver) GetFilter() map[string][]byte {
 func (s *StorageObserver) Listen() {}
 
 // Stop to satisfy Listener interface (but is no longer used by StorageObserver)
-func (s *StorageObserver) Stop() {}
+func (s *StorageObserver) Stop() error { return nil }
 
 // BlockListener to handle listening for blocks importedChan
 type BlockListener struct {
-	Channel chan *types.Block
-	wsconn  WSConnAPI
-	ChanID  byte
-	subID   uint
-
-	ctx    context.Context
-	cancel context.CancelFunc
+	Channel       chan *types.Block
+	wsconn        WSConnAPI
+	ChanID        byte
+	subID         uint32
+	done          chan interface{}
+	cancel        chan interface{}
+	cancelTimeout time.Duration
 }
 
 // Listen implementation of Listen interface to listen for importedChan changes
 func (l *BlockListener) Listen() {
-	l.ctx, l.cancel = context.WithCancel(context.Background())
 	go func() {
+		defer close(l.done)
+		defer close(l.Channel)
```

Comment:
writer should close the channel

Reply:
@EclesioMeloJunior , sorry, will you address this, or should I re-review now? I still stand by this comment, haha